pkgs/development/libraries/fltk/default.nix

```diff
@@ -1,5 +1,199 @@
-import ./common.nix rec {
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  zlib,
+  libjpeg,
+  libpng,
+  fontconfig,
+  freetype,
+  libX11,
+  libXext,
+  libXinerama,
+  libXfixes,
+  libXcursor,
+  libXft,
+  libXrender,
+
+  withGL ? true,
+  libGL,
+  libGLU,
+  glew,
+
+  withCairo ? true,
+  cairo,
+
+  withPango ? false,
+  pango,
+
+  withDocs ? true,
+  doxygen,
+  graphviz,
+
+  withExamples ? (stdenv.buildPlatform == stdenv.hostPlatform),
+  withShared ? true,
+}:
+
+let
+  onOff = value: if value then "ON" else "OFF";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "fltk";
   version = "1.3.8";
-  rev = "release-${version}";
-  sha256 = "1pw4ndwn9rr1cxw5qiw32r9la2z9zbjphgsqq1hj2yy4blwv419r";
-}
+
+  src = fetchFromGitHub {
+    owner = "fltk";
+    repo = "fltk";
+    rev = "release-${finalAttrs.version}";
+    hash = "sha256-OQWyOV3EeyFhwFg/eOX66QtFUxaDR1x4ZyHnZHmzhN8=";
+  };
+
+  outputs = [ "out" ] ++ lib.optional withExamples "bin" ++ lib.optional withDocs "doc";
+
+  # Manually move example & test binaries to $bin to avoid cyclic dependencies on dev binaries
+  outputBin = lib.optionalString withExamples "out";
+
+  patches = lib.optionals stdenv.hostPlatform.isDarwin [
+    ./nsosv.patch
+  ];
+
+  postPatch = ''
+    patchShebangs documentation/make_*
```

Comment:
cmake 4 compat would be nice, e.g. by `substituteInPlace` if upstream doesn't maintain 1.3.x anymore

cmake 4 is currently in `staging-next`, failures and fixes are tracked in https://github.com/NixOS/nixpkgs/issues/445447. The hydra failure for `fltk` can be seen here: https://hydra.nixos.org/build/308206733/nixlog/1